lib/types/btrfs.nix

```diff
@@ -112,26 +116,33 @@ in
     _create = diskoLib.mkCreateOption {
       inherit config options;
       default = ''
-        mkfs.btrfs ${config.device} ${toString config.extraArgs}
-        ${lib.optionalString (config.swap != {}) ''
-          (
-            MNTPOINT=$(mktemp -d)
-            mount ${device} "$MNTPOINT" -o subvol=/
-            trap 'umount $MNTPOINT; rm -rf $MNTPOINT' EXIT
-            ${swapCreate "$MNTPOINT" config.swap}
-          )
-        ''}
-        ${lib.concatMapStrings (subvol: ''
-          (
-            MNTPOINT=$(mktemp -d)
-            mount ${config.device} "$MNTPOINT" -o subvol=/
-            trap 'umount $MNTPOINT; rm -rf $MNTPOINT' EXIT
-            SUBVOL_ABS_PATH="$MNTPOINT/${subvol.name}"
-            mkdir -p "$(dirname "$SUBVOL_ABS_PATH")"
-            btrfs subvolume create "$SUBVOL_ABS_PATH" ${toString subvol.extraArgs}
-            ${swapCreate "$SUBVOL_ABS_PATH" subvol.swap}
-          )
-        '') (lib.attrValues config.subvolumes)}
+        # create the filesystem only if the device seems empty
+        if ! (blkid '${config.device}' -o export | grep -q '^TYPE='); then
+          mkfs.btrfs "${config.device}" ${toString config.extraArgs}
+        fi
+        if (blkid "${config.device}" -o export | grep -q '^TYPE=btrfs$'); then
```

Comment:
I noticed that this is causing ```syntax error near unexpected token `fi'``` when formatting a btrfs filesystem without any swap or subvolume configuration.

I guess this all should be gaurded by `config.swap` & `config.subvolumes` being set? It's a bit annoying bash doesn't support empty if statements.

Reply:
Simplest fix would be `echo -n ""` in the first line.